Dockerfile.spark

```diff
@@ -10,6 +10,8 @@ RUN apt-get update && apt-get install -yq --no-install-recommends --force-yes \
     r-base-core && \
     rm -rf /var/lib/apt/lists/*
 
+RUN pip install --upgrade setuptools
```

Comment:
Why not use ignore installed here too then? Should we update pip here as well?

Reply:
--ignore-installed was not needed here.  I don't know why.  My environment is OSX Sierra with Docker Community Edition Version 17.09.0-ce-mac35 (19611)

As for upgrading pip here..  this image is built via `FROM gettyimages/spark:2.1.0-hadoop-2.7` which had pip up to date.  Adding the upgrade will yield:
Step 3/12 : RUN pip install --upgrade pip
 ---> Running in 58351e04718b
Requirement already up-to-date: pip in /usr/local/lib/python3.4/dist-packages/pip-9.0.1-py3.4.egg
 whereas the jupyter notebook is built via `FROM jupyter/base-notebook:d0b2d159cc6c` and that pinned version is not up to date.

With this image built from an unpinned version of gettyimages, we may be able to defer the pip version to the base docker image.

Reply:
Hi @aggFTW, can you review again?